src/unix_socket.rs

```diff
@@ -0,0 +1,168 @@
+use std::future::Future;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use pin_project_lite::pin_project;
+use reqwest::blocking::{Request, Response};
+use reqwest::header::{HeaderValue, HOST};
+
+pub struct UnixClient {
+    rt: tokio::runtime::Runtime,
+    socket_path: PathBuf,
+    timeout: Option<Duration>,
+}
+
+impl UnixClient {
+    pub fn new(socket_path: PathBuf, timeout: Option<Duration>) -> Self {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        Self {
+            rt,
+            socket_path,
+            timeout,
+        }
+    }
+
+    async fn connect(&self) -> Result<hyper::client::conn::http1::SendRequest<reqwest::Body>> {
+        // TODO: Add support for Windows named pipes by replacing UnixStream with namedPipeClient.
+        // See https://docs.rs/tokio/latest/tokio/net/windows/named_pipe/struct.ClientOptions.html#method.open
+        let stream = tokio::net::UnixStream::connect(&self.socket_path).await?;
+        let (sender, conn) = hyper::client::conn::http1::Builder::new()
+            .title_case_headers(true)
+            .handshake(hyper_util::rt::TokioIo::new(stream))
+            .await?;
+
+        tokio::task::spawn(async move {
+            if let Err(err) = conn.await {
+                log::error!("Connection failed: {:?}", err);
+            }
+        });
```

Comment:
Is it OK to merely log this? I assume it leads to a harder error somewhere down the line and it's just hard to get it out of the task?